iceoryx_hoofs/test/moduletests/test_unit_duration.cpp

```diff
@@ -1785,6 +1847,59 @@ TEST(Duration_test, SubtractDurationWithSecondsAndNanosecondsCausingReductionOfS
     EXPECT_THAT(sut, Eq(EXPECTED_DURATION));
 }
 
+TEST(Duration_test, SubtractAssignSecondsFromDurationResultsInSecondSubtractractionToLHS)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "ab917854-0a97-4529-b408-c51c30b9375f");
+
+    constexpr Duration EXPECTED_DURATION = createDuration(1U, 0U);
+    auto sut = createDuration(2U, 0U);
+    auto otherDuration = createDuration(1U, 0U);
+
+    sut -= otherDuration;
+
+    EXPECT_THAT(sut, Eq(EXPECTED_DURATION));
+}
+
+TEST(Duration_test, SubtractAssignNanosecondsFromDurationResultsInNanosecondSubtractractionToLHS)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "b6051fe8-37d5-4225-b1ea-5038b56889ce");
+
+    constexpr Duration EXPECTED_DURATION = createDuration(0U, 50U);
+    auto sut = createDuration(0U, 100U);
+    auto otherDuration = createDuration(0U, 50U);
+
+    sut -= otherDuration;
+
+    EXPECT_THAT(sut, Eq(EXPECTED_DURATION));
+}
+
+TEST(Duration_test, SubtractAssignDurationPastZeroNanosecondsResultsInDecrementedSeconds)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "cba9f397-9f25-4e88-9608-0d51eb1e4ccc");
+
+    constexpr Duration EXPECTED_DURATION = createDuration(0U, NANOSECS_PER_SECOND - 2);
+    auto sut = createDuration(1U, 0U);
+    auto otherDuration = createDuration(0U, 2U);
+
+    sut -= otherDuration;
+
+    EXPECT_THAT(sut, Eq(EXPECTED_DURATION));
+}
+
+TEST(Duration_test, SubtractAssignDurationPastZeroResultsInZero)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "eb2bace9-751c-48bd-82eb-2e5679512503");
+
+    constexpr Duration EXPECTED_DURATION = createDuration(0U, 0U);
+    auto sut = createDuration(1U, 0U);
+    auto otherDuration = createDuration(2U, 0U);
```

Comment:
Maybe we also want to check that truncation at 0 happens due to nanoseconds as well (i.e. subtract something larger in the nanosecond range from something smaller). Can even be done in the same test IMO.

Reply:
This is tested in the test above: https://github.com/eclipse-iceoryx/iceoryx/pull/1830/files/b45eccf7f12a36dddc0420275ce68c2affff8526#diff-01dd43e4d8b41b14024d0992abde331a2c9b8ec3927488715592516e2e67b8c7R1876

Reply:
@MatthiasKillat